dart/lib/src/sentry_envelope_header.dart

```diff
@@ -0,0 +1,25 @@
+import 'dart:convert';
+
+import 'protocol/sentry_id.dart';
+import 'protocol/sdk_version.dart';
+
+class SentryEnvelopeHeader {
+  SentryEnvelopeHeader(this.eventId, this.sdkVersion);
+  SentryEnvelopeHeader.newEventId()
+      : eventId = SentryId.newId(),
+        sdkVersion = null;
+
+  final SentryId? eventId;
+  final SdkVersion? sdkVersion;
+
+  String serialize() {
+    final serializedMap = <String, dynamic>{};
+    if (eventId != null) {
```

Comment:
If you do it this way, it works with Darts type inference and the compiler knows that you don't need a `!`
```dart
final tempEventId = eventId;
if (tempEventId != null) {
      serializedMap['event_id'] = tempEventId.toString();
}
```